python/ambassador_diag/diagd.py

```diff
@@ -481,11 +486,24 @@ def show_overview(reqid=None):
                  banner_content=banner_content,
                  **ov, **ddict)
 
+    patch_client = request.args.get('patch_client', None)
     if request.args.get('json', None):
         key = request.args.get('filter', None)
 
         if key:
             return jsonify(tvars.get(key, None))
+        elif patch_client:
+            # Get the previous full representation
+            cached_tvars_json = tvars_cache.get(patch_client, dict())
+            # Serialize the tvars into a json-string using the same jsonify Flask serializer, then load the json object
+            response_content = json.loads(flask_json.dumps(tvars))
+            # Diff between the previous representation and the current full representation  (http://jsonpatch.com/)
+            patch = jsonpatch.make_patch(cached_tvars_json, response_content)
+            # Save the current full representation in memory
+            tvars_cache[patch_client] = response_content
```

Comment:
Same comment here. If you save `tvars` instead of `response_content` you should be able to just directly do

```
cached_tvars = tvars_cache.get(patch_client, {})
patch = jsonpatch.make_patch(cached_tvars, tvars)
tvars_cache[patch_client] = tvars
```

no?

Reply:
I experimented with that, but the library is picky and uses a different json encoder than the jsonify one and would spit out serialization errors:
```
TypeError: Object of type datetime is not JSON serializable
```

I played on the safe-side and reused the Flask json encoder for comparing the resuling serialized output vs the dict content.

Reply:
Oh FFS. OK, good enough for now.